Splits the service department, department global var thing II

babydoll opened the pull request:
See #18413

## Service
Staff remaining in service
- Cook
- Bartender
- Janitor
- Crewman

## Exploration
New department, new radio network; retains service dept. uniforms

- SolGov Pilot (retains command, engineering radio)
- Pathfinder (retains command radio)
- Explorer

![](https://s26.postimg.org/c9kp187o9/1505098029.png)

Exploration department radio is on the same server as service. Adding a new server in the telecomms room to keep with the current layout/style would require a remap of the room, and its surrounding areas. Uses unique equipment in Torch folder machinery file to ensure it only works on this map.

Intercom access is set to Heads, as no unique explorer accesses exist at the moment. Once/if explorer offices/equipment is added, this should be replaced with that access.

Also converts existing departments to use global lists for consistency

babydoll committed:
u t i l i t y

atlantiscze commented:
Do we really need yet another radio frequency? There's already way more than comfortable amount of frequencies in my opinion.

As for actual feedback, my partial colourblindness may be affecting this, but the utility radio's colour feels exactly the same as Command's.

babydoll commented:
- There is currently no radio network, other than common, to allow the pathfinder and his staff to communicate
- Other channels do not fit: 
   - Command (they are not heads of staff or supporting heads of staff; there are already many extraneous people on command radio)
   - Engineering (they are not engineers, and their duties do not concern the rest of engineering at all) 
   - Research (most applicable, but they are not NanoTrasen staff)
   - Service (a civilian department; service shouldn't be a catch-all)
- None of these departments really need people talking about moon rocks or whatever on their channel

The colour is a darkish purple. I couldn't find anything else. Command is blue.

AceVX commented:
I don't think we need another radio frequency. If anything, We could rename service to utility since That was meant for "Service staff" of Exodus which does not really apply anymore since its shared by most "other" category roles.

I agree with the logic of having Pilot, Pathfinder and Explorers in their own Frequency. 

But my gripe would be that we already have one agreeably pointless "service" channel. That is almost never used as _primary communication for department operations_ like, say, engineering or command, security and medical are.

Changing service around to utility and piling in the Exploration team + Pilot to it, fixes that issue and defeats a quiet and currently pointless radio frequency by giving it a new name and a real function.

Also, As a moderator my major issue would be how I now have to nigh-indistinguishable colors of Deadchat and Utility radio, at a glance.

babydoll commented:
The colour is easily changed, but I challenge you to find one that isn't in use elsewhere. This is the best compromise I and several other people found. The image here has been there for the past 13 days, and nobody has approached me with a better idea in that time.

On the subject of the content of the PR itself:
I don't think kicking service out of their network is constructive. The XO/HoP should have a way to talk to their staff, all of which are in the service department and channel. I use(d) it all the time when I play(ed) XO.

Keeping them in service and giving them service radio would mean that there's an **explicitly** military-only team (pilot, pathfinder, explorers) discussing mission-critical issues on the same network as potentially civilian low-level staff like the janitor.

I value everyone's feedback, but I wouldn't have gone though all the work of doing it, fixing issues and fixing conflicts over and over if I didn't think it was a good, constructive change. 

I feel like these slots were added too soon (cf. pilot, exoplanets), but this is going to become more necessary as exploration content is added.

Techhead0 commented:
I was in favor of giving them the Science channel. It's labeled "Science", not "NanoTrasen", and NT staff are who they'll be most often needing to talk to anyway.

babydoll commented:
Updated the colour in the OP.  I think it's a fair assessment to say that the research channel should be for NT staff only. The only army guy who has it is the captain. Not even the XO gets it. I don't want to start introducing rank and file military people into it.

AtomicWorm commented:
Obligatory "The Expeditionary Corps are not military" post.

I agree with you, though. But, with that new color change, what will be their actual new "department color"? Will it remain purple?

PurpleMartinJCK commented:
Yeah I think giving them a science key is a better solution for it. Or making a channel they and NT peeps use while on away missions specifically.

BisLordLoko commented:
I just think it's a little bit weird calling it "Ultility", wouldn't be better calling it "Exploration"?

babydoll commented:
@FTangSteve et al
I can't see why this new network is apparently so shocking. We already added the slots, and they have barely anything to do. We added the pilot slots when they had almost nothing to do. This is future proofing for when the pathfinder and explorers (and others, if/when they're added) need to conduct a reasonably private conversation, amongst themselves, on matters that are not suitable for other channels. 

The _Research_ radio is specifically for staff in the _Research_ Department, just as the security, medical, engineering and supply radios are for members of those departments. Service is the aberration in that they have two distinct (in my view) departments sharing the same radio network (or they would, if the exploration team had any departmental radios at all) and department categorization. I have an _**E**xcellent **T**rack **R**ecord™_ when it comes to department categorization, which the continued success of the command support department can attest to.

I think a reasonable compromise would be to add the utility network to the RD and optionally senior researcher's radios. I don't personally think this is the best idea since inevitably someone is going to start acting like they're someone else's boss when it isn't the case, but I can see your point in that the utility team sort of uniquely relies on the research team.

**e**: If any developer, or anyone else, feels as though I haven't addressed their concerns so far, please let me know here or on IRC. I've invested some effort into this, so I'm pretty determined to see it through.

babydoll committed:
Merge branch 'dev' of https://github.com/Baystation12/Baystation12 into ugehrugheutility

# Conflicts:
#	code/_helpers/global_access.dm
#	test/run-test.sh

babydoll commented:
Regarding the name, chinsky said that he has a PR to change the name in the works, if this is merged. I believe someone is working on uniforms, as well.

AtomicWorm commented:
That's me. Which brings me to a request. The department color is kinda close to command's. It's more of a dark yellow as compared to command's mixture of gold and urine, but with 32x32 sprites, they could get confused. I don't suppose I could plead with you for a more distinct color?

babydoll commented:
The radio colour doesn't need to correspond to the uniform colour, pick whatever is available that looks nice. IMO.

babydoll commented:
After a lot of feedback, I've changed the name from Utility to Exploration.

PurpleMartinJCK commented:
Ok so my thoughts on this remain that I don't like the idea of giving them a new channel just for them that the other people who will be going on the expedition will not be able to use. I feel this will promote then communicating amongst themselves rather than working as a group. One solution for this would be to add encryption keys for the channel into research lockers and onto the mission ships so other people could hook in to the channel.

babydoll commented:
Your suggestion runs directly counter to the objective of this PR, which is to introduce a new department and a secure department radio network for that department. Yes, this will promote them communicating amongst themselves. _That is the point of a department radio network._

The shared radio network on the away sites is known as 'common'. It is accessed through the hand radios. This is the department radio network for the exploration department. It is not intended to be a shared network. It is the department radio network for the exploration department, just like every other department has.

There's no way to enforce the research staff not abusing the free keys to always have access to this channel. It's like if we put security encryption keys in every locker in the game and then expected people not to use them. It's unrealistic to expect that they won't. Medical doesn't have security or engineering, when both those networks would be much more useful to them than exploration to research or vise versa.

Planning that doesn't occur in person should be going through the supervisors for the departments (pathfinder/XO, RD). This gives these people something to do. It's why they're here. It's why the command network is shared by all the departments.

babydoll committed:
Merge branch 'dev' into ugehrugheutility

babydoll committed:
Update jobs.dm

PurpleMartinJCK commented:
If it does get through, the wiki will need to be updated accordingly.

babydoll commented:
Yes of course. This remains ready for review

Techhead0 reviewed (approved):
I don't like the concept, but the code looks good.
Consider this a begrudging approval of the PR itself.

PsiOmegaDelta commented:
I'll merge this just so I can begin working on throwing out current radio and department code.
This is a horrible mess.

PurpleMartinJCK commented:
Heya, so this went in, did you update the wiki?

babydoll commented:
I went to look and someone had already made the changes I was going to on the 20th.

PurpleMartinJCK commented:
Can you post them here please?

babydoll commented:
I was going to create blank pages for explorer, pathfinder and the exploration department. Those are already done. I was going to add the department to the jobs page itself, which has already been done. I was going to edit the headset page to reflect the new channel, but that page doesn't exist.

I've never played any of these jobs, so I'm not the person to ask to write content for them.

PurpleMartinJCK commented:
Did you check the radio pages and update the talk keys for the new channel? You can write a paragraph onthe radio etc.

babydoll commented:
If such a page exists, I can't find it. Message me on IRC and tell me what you want me to do, and I'll tell you if I'll do it or not.

PurpleMartinJCK commented:
Wiki stuff is good. You're good to go.
